src/analytics_desktop.cc

```diff
@@ -0,0 +1,405 @@
+    } else if (param.value.is_vector()) {
+      // Vector types for top-level event parameters are not supported on Desktop.
+      // Only specific complex types (like a map processed into an ItemVector) are handled.
+      LogError("Analytics: Parameter '%s' has type Vector, which is unsupported for event parameters on Desktop. Skipping.", param.name);
+      continue; // Skip this parameter
+    } else if (param.value.is_map()) {
+      // This block handles parameters that are maps.
+      // Each key-value pair in the map is converted into a GoogleAnalytics_Item,
+      // and all such items are bundled into a GoogleAnalytics_ItemVector,
+      // which is then inserted into the event parameters.
+      // The original map's key becomes the "name" property of the GA_Item,
+      // and the map's value becomes one of "int_value", "double_value", or "string_value".
+      const std::map<std::string, firebase::Variant>& user_map =
+          param.value.map_value();
+      if (user_map.empty()) {
+        LogWarning("Analytics: Parameter '%s' is an empty map. Skipping.", param.name);
+        continue; // Skip this parameter
+      }
+
+      GoogleAnalytics_ItemVector* c_item_vector =
+          GoogleAnalytics_ItemVector_Create();
+      if (!c_item_vector) {
+        LogError("Analytics: Failed to create ItemVector for map parameter '%s'.", param.name);
+        continue; // Skip this parameter
+      }
+
+      bool item_vector_populated = false;
+      for (const auto& entry : user_map) {
+        const std::string& key_from_map = entry.first;
+        const firebase::Variant& value_from_map = entry.second;
+
+        GoogleAnalytics_Item* c_item = GoogleAnalytics_Item_Create();
+        if (!c_item) {
+          LogError("Analytics: Failed to create Item for key '%s' in map parameter '%s'.", key_from_map.c_str(), param.name);
+          continue; // Skip this key-value pair, try next one in map
+        }
+
+        // Removed: GoogleAnalytics_Item_InsertString(c_item, "name", key_from_map.c_str());
+
+        bool item_had_valid_property = false;
+        if (value_from_map.is_int64()) {
+          GoogleAnalytics_Item_InsertInt(c_item, key_from_map.c_str(), value_from_map.int64_value());
+          item_had_valid_property = true;
+        } else if (value_from_map.is_double()) {
+          GoogleAnalytics_Item_InsertDouble(c_item, key_from_map.c_str(), value_from_map.double_value());
+          item_had_valid_property = true;
+        } else if (value_from_map.is_string()) {
+          GoogleAnalytics_Item_InsertString(c_item, key_from_map.c_str(), value_from_map.string_value());
+          item_had_valid_property = true;
+        } else {
+          LogWarning("Analytics: Value for key '%s' in map parameter '%s' has an unsupported Variant type. This key-value pair will be skipped.", key_from_map.c_str(), param.name);
+        }
+
+        if (item_had_valid_property) {
```

Comment:
This can be cleaned up to avoid the bool, just do an else at the end of the if tree.